contrib/lib/src/databases.rs

```diff
@@ -0,0 +1,209 @@
+///
+/// Example configuration:
+/// ```toml
+/// [[global.databases]]
+/// name = "my_database"
+/// url = "postgres://postgres@localhost/my_database
+/// pool_size = 4
+/// ```
+pub struct DatabaseConfig<'a> {
+    /// The connection URL specified in the Rocket configuration
+    pub url: &'a str,
+    /// The size of the pool to be initialized. Defaults to the number of Rocket workers.
+    pub pool_size: i64,
+    /// Any extra options that are included in the configuration.
+    pub extras: &'a BTreeMap<String, Value>,
+}
+
+/// A custom error type to wrap multiple, disparate error types
+///
+/// This allows for returning the given error from the connection manager
+/// creation methods and the R2D2 connection pool methods.
+#[derive(Debug)]
+pub enum DbError<T> {
+    Custom(T),
+    PoolError(r2d2::Error),
+}
+
+#[derive(Debug)]
+pub enum DatabaseConfigError<'a> {
+    MissingConfig,
+    MissingTable(&'a str),
+    MissingConnectionString(&'a str),
+    InvalidPoolSize(i64),
+}
+
+impl<'a> DatabaseConfigError<'a> {
+    pub fn pretty_print(&self) {
+        error!("{}", self);
+    }
+}
+
+impl<'a> Display for DatabaseConfigError<'a> {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self {
+            DatabaseConfigError::MissingConfig => {
+                write!(f, "Could not find the `databases` key in the configuration")
+            },
+            DatabaseConfigError::MissingTable(table_name) => {
+                write!(f, "Could not find the configuration named `{}`", table_name)
+            },
+            DatabaseConfigError::MissingConnectionString(conn_string) => {
+                write!(f, "The specified connection string, `{}`, is invalid", conn_string)
+            },
+            DatabaseConfigError::InvalidPoolSize(pool_size) => {
+                write!(f, "The specified pool size, `{}`, is invalid", pool_size)
+            },
+        }
+    }
+}
+
+/// This method retrieves the database configuration from the `Rocket.toml` file
+/// and returns a DatabaseConfig struct.
+///
+/// Example usage:
+/// ```rust,ignore
+/// let config = database_config("my_database", rocket.config());
+/// ```
+pub fn database_config<'a>(
+    name: &'a str,
+    from: &'a config::Config
+) -> Result<DatabaseConfig<'a>, DatabaseConfigError<'a>> {
+    // Find the first `databases` config that's a table with a key of 'name'
+    // equal to `name`.
+    let connection_config = from.get_slice("databases")
+        .map_err(|_| DatabaseConfigError::MissingConfig)?
+        .iter()
+        .filter_map(|config| config.as_table())
+        .filter(|table| table.get("name").map(|n| n.as_str() == Some(name)).unwrap_or(false))
+        .next()
+        .ok_or(DatabaseConfigError::MissingTable(name))?;
+
+    let url = connection_config.get("url")
+        .and_then(Value::as_str)
+        .ok_or(DatabaseConfigError::MissingConnectionString("Could not fetch a valid connection string for the database configuration"))?;
```

Comment:
Why store this string inside of the `MissingConnectionString` variant? Why not simply `MissingUrl` or something?